cmd/diff.go

```diff
@@ -0,0 +1,197 @@
+	goc diff --prow-postsubmit-job=<xxx> --prow-remote-profile-name=<xxx> 
+    		 --qiniu-credential=<xxx> --new-profile=<xxxx> // diff local coverage profile with the remote one in prow job
+	goc diff --prow-postsubmit-job=<xxx> --prow-profile=<xxx> 
+    		 --github-token=<xxx> --github-user=<xxx> --github-comment-prefix=<xxx> 
+    		 --qiniu-credential=<xxx> --coverage-threshold-percentage=<xxx> --new-profile=<xxxx> // diff coverage profile with the remote one in prow job, and post comments to github PR
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if baseProfile != "" {
+			doDiffForLocalProfiles(cmd, args)
+		} else if prowPostSubmitJob != "" {
+			doDiffUnderProw(cmd, args)
+		} else {
+			logrus.Fatalf("either base-profile or prow-postsubmit-job must be provided")
+		}
+	},
+}
+
+var (
+	newProfile        string
+	baseProfile       string
+	coverageThreshold int
+
+	prowPostSubmitJob string
+	prowProfile       string
+
+	githubToken         string
+	githubUser          string
+	githubCommentPrefix string
+
+	qiniuCredential string
+
+	robotName string
+	fullDiff  bool
+)
+
+func init() {
+	diffCmd.Flags().StringVarP(&newProfile, "new-profile", "n", "", "local profile which works as the target to analysis")
+	diffCmd.MarkFlagRequired("new-profile")
+	diffCmd.Flags().StringVarP(&baseProfile, "base-profile", "b", "", "another local profile which works as baseline to compare with the target")
+	diffCmd.Flags().IntVarP(&coverageThreshold, "coverage-threshold-percentage", "", 0, "coverage threshold percentage")
+	diffCmd.Flags().StringVarP(&prowPostSubmitJob, "prow-postsubmit-job", "", "", "prow postsubmit job which used to find the base profile")
+	diffCmd.Flags().StringVarP(&prowProfile, "prow-remote-profile-name", "", "filtered.cov", "the name of profile in prow postsubmit job, which used as the base profile to compare")
+	diffCmd.Flags().StringVarP(&githubToken, "github-token", "", "/etc/github/oauth", "path to token to access github repo")
+	diffCmd.Flags().StringVarP(&githubUser, "github-user", "", "", "github user name when comments in github")
+	diffCmd.Flags().StringVarP(&githubCommentPrefix, "github-comment-prefix", "", "", "specific comment flag you provided")
+	diffCmd.Flags().StringVarP(&qiniuCredential, "qiniu-credential", "", "/etc/qiniuconfig/qiniu.json", "path to credential file to access qiniu cloud")
+	diffCmd.Flags().StringVarP(&robotName, "robot-name", "", "qiniu-bot", "github user name for coverage robot")
+	diffCmd.Flags().BoolVarP(&fullDiff, "full-diff", "", false, "when set true,calculate and display full diff coverage between new-profile and base-profile")
+
+	rootCmd.AddCommand(diffCmd)
+}
+
+//goc diff --new-profile=./new.cov --base-profile=./base.cov
+//+------------------------------------------------------+---------------+--------------+--------+
+//|                         FILE                         | BASE COVERAGE | NEW COVERAGE | DELTA  |
```

Comment:
Both codecov and coverage-bot not use all uppercase in the header， here i suggest we'b better follow that rule.